lib/broadway_kafka/brod_client.ex

```diff
@@ -147,6 +153,20 @@ defmodule BroadwayKafka.BrodClient do
     end
   end
 
+  @impl true
+  def prepare_for_start(broadway_opts) do
+    {_, producer_opts} = broadway_opts[:producer][:module]
+    init_opts = Keyword.put(producer_opts, :broadway, broadway_opts)
+
+    case init(init_opts) do
+      {:error, message} ->
+        raise ArgumentError, "invalid options given to #{__MODULE__}.init/1, " <> message
```

Comment:
I think this module is not quite public, so I would rather include the user module instead in the error message. like invalid options were given to Broadway.start_link.

Reply:
I have implemented the same error message that is being raised currently on https://github.com/dashbitco/broadway_kafka/blob/3227bf0bfc31ed60d9d10192256cc36dc3c95afd/lib/broadway_kafka/producer.ex#L238-L239

Should I update both places?